components/admin/reports/GenerateReportModal.tsx

```diff
@@ -157,102 +191,76 @@ export default function GenerateReportModal<T>(props: GenerateReportProps<T>) {
               <Checkbox
                 paddingBottom="6px"
                 fontWeight="bold"
-                isChecked={selectAll}
-                onChange={event => {
-                  setSelectAll(event.target.checked);
-                  setUserID(event.target.checked);
-                  setApplicantName(event.target.checked);
-                  setApplicantDoB(event.target.checked);
-                  setAPPNumber(event.target.checked);
-                  setApplicationDate(event.target.checked);
-                  setPaymentMethod(event.target.checked);
-                  setFeeAmount(event.target.checked);
-                  setDonationAmount(event.target.checked);
-                  setTotalAmount(event.target.checked);
-                }}
+                isChecked={
+                  page === 'requests'
+                    ? selectedColumns.has(ApplicationsReportColumn.UserId) &&
```

Comment:
Let's move this to a memoized variable outside the JSX for readability and performance improvement. We can also iterate through the TS enum as so:
```typescript
import { useMemo } from 'react'

const areAllColumnsSelected = useMemo(() => Object.keys(ApplicationsReportColumn).every(column => selectedColumns.has(column)), [selectedColumns])
```

Reply:
Neat! Addressed!